.github/workflows/security.yaml

```diff
@@ -32,7 +35,60 @@ jobs:
           api-key: ${{ secrets.FOSSA_API_KEY }}
 
       - name: Run FOSSA Test
-        uses: fossas/fossa-action@main
-        with:
-          api-key: ${{ secrets.FOSSA_API_KEY }}
-          run-tests: true
+        id: fossa_test
+        env:
+          FOSSA_API_KEY: ${{ secrets.FOSSA_API_KEY }}
+        run: |
+          MAIN_BRANCH="refs/heads/main"
+          CURRENT_BRANCH="${{ github.ref }}"
+          DIFF_OPTION=""
+      
+          if [ "$CURRENT_BRANCH" != "$MAIN_BRANCH" ]; then
+            git fetch origin main
+            MAIN_SHA=$(git rev-parse origin/main)
+            DIFF_OPTION="--diff $MAIN_SHA"
+          fi
+      
+          fossa test --format json $DIFF_OPTION | tee fossa-test.json 1>&2
```

Comment:
what's going on at the end here? Why do we want to output this to stderr (if i'm reading this right)

Reply:
nice catch, I tested it and this is not needed... It was an attempt to swallow exit code 1